FEMComputationalFluidDynamicsAnalysis/postProcessing/computeResultingTotalForceOnSelectedDomain.m

```diff
@@ -0,0 +1,39 @@
+function F = computeResultingTotalForceOnSelectedDomain(analysis, nodesDomain, FComplete, parameters)
+%% Licensing
+%
+% License:         BSD License
+%                  cane Multiphysics default license: cane/license.txt
+%
+% Main authors:    Marko Leskovar
+%                  Andreas Apostolatos
+%
+%% Function documentation
+%
+% Calculate force on a single body
+%
+%              Input :
+%           analysis : Analysis type and number of dimensions
+%        nodesDomain : Global numbering of the nodes on the selected domain
+%          FComplete : The complete force vector
+%         parameters : Flow parameters
+%   
+%             Output :
+%                  F : Vector of noSpatialDimensions containing the 
+%                      components of the total resulting force along each 
+%                      spatial dimension over the selected domain
+%
+%% Function main body
+    
+% Reshape the vector to [Vx, Vy, p]
+FComplete = -reshape(FComplete, [analysis.noFields,                     ...
+             length(FComplete) / analysis.noFields])';
+
+% Select only the forces on the domain
+Force = FComplete(nodesDomain, 1:analysis.noSpatialDimensions);
+
+F = zeros(analysis.noSpatialDimensions,1);
+for k = 1:analysis.noSpatialDimensions
+    F(k,1) = sum(Force(:,k)) * parameters.rho;
+end
+    
```

Comment:
The aforementioned lines of code can be here spared and moreover lead to a more readable code. I would propose to try the following,

F = zeros(analysis.noSpatialDimensions,1);
￼for i = 1:analysis.noSpatialDimensions
￼    F(i,1) = sum(FComplete(k:analysis.noFields:end,1)) * parameters.rho;
￼end

this should give the exact same result but in a much fewer and more efficient manner. This is because you spare creating new variables and reshaping them into FComplete = [Fx, Fy, Fz]. 

Command FComplete(k:analysis.noFields:end,1) returns the components of FComplete with indiced [k   k + analysis.noFields    ...] which for k = 1 would be all x-components of the residual force and for k = 2 all y-components of the residual force.

Reply:
% The DOFs in the domain of interest
DOFsDomain = sort([3*nodesDomain - 2 3*nodesDomain - 1]);

% Compute the resultant reaction forces over the domain of interest along each Cartesian direction
F = zeros(analysis.noSpatialDimensions,1);
for i = 1:analysis.noSpatialDimensions
F(i,1) = sum(FComplete(DOFsDomain(i:analysis.noFields:end),1))*parameters.rho;
end

Reply:
This sadly didn't work for 3D cases as one would have to expand:
"DOFsDomain = sort([3 * nodesDomain - 2 3 * nodesDomain - 1])"
with one futher entry  " 3 * nodesDomain - 1" to take into accound more fields. 

But I implemented something realy similar that works for 2D and 3D cases.

Best Regards,
Marko Leskovar